src/documents/PromisesAPI.md

```diff
@@ -0,0 +1,236 @@
+promises are in play in a given piece of code. Secondly, our implementation will synchronously relay a value which is available
+synchronously - this means that in the debugger, or other source of stack traces, the maximal size of stack will be visible to 
+account for the cause of the promise resolution.
+
+Plenty of arguments exist against these choices - in fact, these choices place us firmly in the category of people who "don't really understand
+promises and think of them as glorified ... callback aggregators". In the meantime, we have work to do. Infusion is about the elimination of
+code, and so we only have limited time to spend thinking about how to make the code we do have conform to a faulty ideal of what we dreamed
+that the virtues of conventional, synchronous code might once have been. However, it's worth noting that there is at least [one virtue](#promise-accumulaterejectionreason-error-) of 
+conventional, synchronous code that is recaptured by no other promise system but ours. Also, this [github issue](https://github.com/promises-aplus/promises-spec/issues/4) attached to the 
+A+ promises specification is a useful source of convincing argumentation that the decision in favour of universally asynchronous resolution is flawed.
+
+As a further landmark for discussion, note that in terms of the following very illuminating [category theoretic](http://brianmckenna.org/blog/category_theory_promisesaplus) treatment of 
+promises (itself rejected by mainstream promises proponents), our `then` method is very definitely ***not*** "the name for `flatMap`". 
+Our `then` method is simply a "glorified callback aggregator".
+
+The implementation skeleton for Infusion's promises was taken from a code sample by John Hann (unscriptable) in this [gist](https://gist.github.com/unscriptable/814052) - full credit and thanks,
+and please read the gist for some further commentary and coverage of limitations.
+
+
+## Core Promises Utility API
+
+The library implements a few utilities without which it is inconvenient to use promises:
+
+### fluid.isPromise(totest)
+
+* `totest {Any}` An object to be checked for being a promise
+* Returns: `{Boolean}` If `totest` has a member `then` of type `Function`, returns `true`.
+
+Determines whether an object is a promise, for our purposes. Any object with a member `then` of type `Function` passes this test.
+This includes essentially every known variety, including jQuery promises. This test in fact identifies what in other libraries is
+termed a "foreign thenable".
+
+### fluid.toPromise(value)
+
+* `value {Any}` A value to be converted ("hoisted") to a promise
+* Returns: `{Promise}` If the supplied value is already a promise, it is returned unchanged. Otherwise a fresh promise is created with the value as resolution and returned.
+
+Coerces any value to a promise. If it is already a promise, it is returned unchanged. Otherwise, `toPromise` constructs a fresh promise resolving to the supplied value.
```

Comment:
This paragraph basically says the same thing as the description of the return value. This section could be tightened up a bit to avoid this redundancy.